### PR TITLE
Update Tailwind to v3

### PR DESCRIPTION
**Pull request by krischarbonneau**
### Description

The purpose of this PR is to upgrade Tailwind to v3 following [the official upgrade guide](https://tailwindcss.com/docs/upgrade-guide). I've also added the tailwind typography plugin.

List of proposed changes:

- We could explore adding some default custom colors that align with the Canada.ca theme 

### What to test for/How to test

1. Pull in branch
2. Try adding tailwind classes to an element on `index.js`
3. Ensure styling still works

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 module.exports = {
-  mode: 'jit',
-  purge: ['./pages/**/*.{js,ts,jsx,tsx}', './components/**/*.{js,ts,jsx,tsx}'],
-  darkMode: false,
+  content: [
+    './pages/**/*.{js,ts,jsx,tsx}',
+    './components/**/*.{js,ts,jsx,tsx}',
+  ],
   theme: {
+    colors: {
```
Comment (shewood) [inline tailwind.config.js:7]: Do we want to completely replace all of the tailwind colours? I didn't think we did

Comment (krischarbonneau) [reply tailwind.config.js:7]: Good catch, didn't realize I hadn't put this inside `extend`, which also explains the issue I was having with `ring-black` in `globals.css`

Comment (P0NDER0SA) [reply tailwind.config.js]: you fixed this, @krischarbonneau ?

Comment (krischarbonneau) [reply tailwind.config.js]: yes, this was resolved with my latest commit @P0NDER0SA

**Review by shewood — APPROVED** [review]
Looks good to me now.

**Pull request merged**